File: app/src/main/java/com/chocolateloverraj/androidapiwebview/MainActivity.java
package com.chocolateloverraj.androidapiwebview;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private TextView openText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        openText = findViewById(R.id.openText);
    }

    public void onOpen(View view) {
        System.out.println(openText.getText());
        Intent intent = new Intent(this, PageViewer.class);
        Uri uri = Uri.parse(openText.getText().toString());
        if (uri.getScheme() == null)
            uri = Uri.fromParts("android-api-webview", uri.getPath(), uri.getFragment());
        else if (!uri.getScheme().equals("https")) {
            Toast.makeText(this, "Url must be https", Toast.LENGTH_LONG).show();
            return;
        }
        System.out.println(uri);
        System.out.println(uri.getScheme() + ", " + uri.getHost() + ", " + uri.getPath() + ", " + uri.getFragment());
        intent.setData(uri);
        startActivity(intent);
    }
}

File: app/src/main/java/com/chocolateloverraj/androidapiwebview/MainActivity.kt
package com.chocolateloverraj.androidapiwebview

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.Navigation
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import com.chocolateloverraj.androidapiwebview.databinding.ActivityMain2Binding
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    private var binding: ActivityMain2Binding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMain2Binding.inflate(layoutInflater)
        setContentView(binding!!.root)
        findViewById<BottomNavigationView>(R.id.nav_view)
        // Passing each menu ID as a set of Ids because each
        // menu should be considered as top level destinations.
        AppBarConfiguration.Builder(
                R.id.navigation_home, R.id.navigation_dashboard)
                .build()
        val navController = Navigation.findNavController(this, R.id.nav_host_fragment_activity_main2)
        NavigationUI.setupWithNavController(binding!!.navView, navController)
    }
}